drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -352,8 +355,11 @@ static int i2c_mspm0g3xxx_transmit(const struct device *dev, struct i2c_msg msg,
 	while ((DL_I2C_getControllerStatus((I2C_Regs *)config->base) &
 		 DL_I2C_CONTROLLER_STATUS_BUSY))
 		;
-	DL_I2C_startControllerTransfer((I2C_Regs *)config->base, data->addr,
-				       DL_I2C_CONTROLLER_DIRECTION_TX, data->msg.len);
+	DL_I2C_startControllerTransferAdvanced((I2C_Regs *)config->base, data->addr,
+		DL_I2C_CONTROLLER_DIRECTION_TX, data->msg.len,
+		DL_I2C_CONTROLLER_START_ENABLE,
+		(msg.flags & I2C_MSG_STOP) ? DL_I2C_CONTROLLER_STOP_ENABLE : DL_I2C_CONTROLLER_STOP_DISABLE,
+		DL_I2C_CONTROLLER_ACK_ENABLE);
```

Comment:
When I looked at the TI HAL stuff, I couldn't quite understand whether to use ACK_ENABLE or ACK_DISABLE for various transfers.
Did you find something specific about when to use one or the other?

Reply:
The basic transfer is masking this bit away. So the register just maintains the current state. I was thinking of making a custom function to not touch the ack bit. But I refrained from doing it since acknowledgements need to be enabled. Note: I noticed that I am disabling them for RX, will fix.

Reply:
I have updated the patch to leave the ack bit alone and only configure the stop bit.